ddl/db_partition_test.go

```diff
@@ -1449,3 +1450,54 @@ func (s *testIntegrationSuite4) TestPartitionErrorCode(c *C) {
 	_, err = tk.Exec("alter table t_part coalesce partition 4;")
 	c.Assert(ddl.ErrCoalesceOnlyOnHashPartition.Equal(err), IsTrue)
 }
+
+func (s *testIntegrationSuite5) TestConstAndTimezoneDepent(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	// add partition
+	tk.MustExec("set @@session.tidb_enable_table_partition = 1")
+	tk.MustExec("drop database if exists test_db_with_partition_const")
+	tk.MustExec("create database test_db_with_partition_const")
+	tk.MustExec("use test_db_with_partition_const")
+
+	sql1 := `create table t1 ( id int ) 
+		partition by range(4) (
+		partition p1 values less than (10)
+		);`
+	assertErrorCode(c, tk, sql1, tmysql.ErrWrongExprInPartitionFunc)
+
+	sql2 := `create table t2 ( time_recorded timestamp ) 
+		partition by range(TO_DAYS(time_recorded)) (
```

Comment:
TO_DAYS function works on datetime type? 

How about adding this one:
```
create table t2 ( time_recorded datetime ) 
		partition by range(TO_DAYS(time_recorded)) ( 
...
```

Reply:
Please check the error code.
It does not work on date time type, i will check it in next line
```
assertErrorCode(c, tk, sql2, tmysql.ErrWrongExprInPartitionFunc)
```

Reply:
> I mean, how about adding this line:
> 
> ```
> tk.MustExec("create table t2 ( time_recorded datetime ) 
> 		partition by range(TO_DAYS(time_recorded)) ( 
> ...")
> ```
> 
> If my memory serve me right, there are some table partition tests use `to_days` as partition function
> https://github.com/pingcap/tidb-test/blob/master/mysql_test/t/partition_range.test#L17

This sql is right, the rules please check the description or mysql comments.
Thanks

Reply:
> https://github.com/pingcap/tidb-test/blob/master/mysql_test/t/partition_range.test#L17

For `to_days` method, adding following cases
1. to_days(time)
2. to_days(timestamp)
3. to_days(datetime)
4. to_days(date)